gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -948,11 +962,13 @@ MM_MemoryPoolLargeObjects::contractWithRange(MM_EnvironmentBase* env, uintptr_t
 		/* No LOA so just remove memory from SOA and we are done */
 		_memoryPoolSmallObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
 	} else {
-
-		assume0(lowAddress >= _currentLOABase);
-
-		/* First remove memory from LOA */
-		_memoryPoolLargeObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
+		if (lowAddress >= _currentLOABase) {
+			_memoryPoolLargeObjects->contractWithRange(env, contractSize, lowAddress, highAddress);
+		} else {
+			Assert_MM_true(getCurrentLOASize() == getApproximateFreeLOAMemorySize());
+			_memoryPoolLargeObjects->contractWithRange(env, getCurrentLOASize(), _currentLOABase, highAddress);
+			_memoryPoolSmallObjects->contractWithRange(env, contractSize - getCurrentLOASize(), lowAddress, (void*)(((uintptr_t)highAddress) - _loaSize));
```

Comment:
I'm curious how these two contracts show up in verbose GC

Reply:
I guess after these 2 contractions, LOA will go away. I think it would be wise to re-create LOA again. Could you try it?

Reply:
The call to redistributeFreeMemory(...) on the next line there should handle that I think

Reply:
good point, I missed it

still, I'd like to see some evidence (example) of this path being taken, which we should be able to identify with VGC . use tricks like narrowing gap between minf maxf to increase probability of contractions, if needed.